### PR TITLE
two last commits now also in develop branch

### PR DESCRIPTION
**Pull request by TheTesla**
I thought development is done on master branch. Now it seems everything is done in develop. So I did my changes also in this branch.

### CODE REVIEW

File: scripts/systemd/bumblebeed.service.in [hunk 1]
```diff
@@ -3,6 +3,7 @@ Description=Bumblebee C Daemon
 
 [Service]
 Type=simple
+ExecStartPre=/usr/bin/prime-select intel
```
Comment (bluca) [inline scripts/systemd/bumblebeed.service.in]: This should be:

```
ExecStartPre=-/usr/bin/prime-select intel
```

As prime-select is only available on certain distributions, not everywhere.

File: src/findpathwild.c [hunk 2]
```diff
@@ -0,0 +1,206 @@
+/***********************************************************************
```
Comment (bluca) [inline src/findpathwild.c]: Please use the same GPL headers as other source files

**Comment by TheTesla** [comment]
I did the changes you asked me to do. You can merge it now.

**Comment by bluca** [comment]
Looks fine for me, even though it's just to work around a limitation in the Ubuntu drivers, but @Lekensteyn and @ArchangeGabriel need to approve

**Comment by TheTesla** [comment]
I have the prime-select intel only in scripts/systemd/bumblebeed.service.in
It would be nice, if someone with knowledge about sysvinit and upstart could do it for this.

**Comment by Lekensteyn** [comment]
The old Upstart script for Ubuntu detected the drivers right here:
https://github.com/Bumblebee-Project/bumblebee-ppa/blob/master/xenial/bumblebee/debian/bumblebee.bumblebeed.upstart

For systemd.. hmm, that has never been updated.

The proposed patch has some potential issues:
- memory leaks (malloc not being freed)
- not checking result of malloc
- potential deep recursion if for some reason (e.g. custom kernels) you have symlinks to outside the directory, e.g. /lib/modules/XXX/source -> /tmp/source

Rather than modifying the C source code to accomodate an Ubuntu-specific problem, what about a custom systemd configuration for Ubuntu which does the equivalent of the upstart script?

It could even be a drop-in file in `/lib/systemd/system/bumblebeed.service.d/`, something like:

```
[Service]
ExecStart=
ExecStart=/usr/bin/bumblebeed.wrap
```

with `bumblebeed.wrap` being this Ubuntu-specific script that detects the path and passes corresponding bumblebeed options. Or alternatively play with EnvironmentFile, see https://lists.freedesktop.org/archives/systemd-devel/2012-October/007273.html

Sketch of the idea (untested, creating bumblebee-detect-nvidia is an exercise for the reader):
```
# /lib/systemd/system/bumblebeed.service.d/ubuntu-nvidia-detect.conf
[Service]
# Load driver options (if any), use a reasonable default
Environment=PKGNAME=nvidia
EnvironmentFile=-/run/bumblebeed.options

# Disable previous ExecStart command
ExecStart=
# run script that writes detected driver to a file
ExecStartPre=-/usr/bin/bumblebee-detect-nvidia
# Start bumblebeed with detected nvidia driver
ExecStart=/usr/bin/bumblebeed --driver nvidia --driver-module "$PKGNAME" --ldpath "/usr/lib/$PKGNAME:/usr/lib32/$PKGNAME" --module-path "/usr/lib/$PKGNAME/xorg,/usr/lib/xorg/modules"
```

**Comment by TheTesla** [comment]
@Lekensteyn: Your solution looks better. At first I searched for a solution not modifying the bumblebeed itself. I had not noticed the cli options. I also didn't know it is done already for upstart or sysvinit. I think I will write a bash or python script doing the wildcard paths (or automatic detection) for systemd.

**Comment by TheTesla** [comment]
@Lekensteyn: Please look at Pull Request #901. This is done with a bash script, not touching the executable itself.

**Comment by Lekensteyn** [comment]
Superseded by #901

**Pull request closed without merging**